sql/core/pom.xml

```diff
@@ -291,5 +297,68 @@
         </dependency>
       </dependencies>
     </profile>
+    <profile>
```

Comment:
I think we should also shaded and relocation `protobuf-java` in sql module, like 

https://github.com/LuciferYang/spark/blob/ef9c8e0d045576fb325ef337319fe6d59b7ce858/core/pom.xml#L691-L694

Reply:
Would you mind adding to this pr? This is some necessary initial work @techaddict 

both `pom.xml` and `SparkBuild.scala`

Reply:
for SBT we can refer to 

https://github.com/apache/spark/blob/73593d8670ca5dd933327039fc659dd6ef764963/project/SparkBuild.scala#L618-L631

https://github.com/apache/spark/blob/73593d8670ca5dd933327039fc659dd6ef764963/project/SparkBuild.scala#L645-L654

Otherwise, GA task may can't compile proto

Reply:
The old protobuf deps in root pom causes so many troubles...

Reply:
https://github.com/apache/spark/blob/e56f31d5e764cd68afed87675e63be7f4fff3e54/pom.xml#L821-L831

From the comments, maybe we can try to remove the old protobuf dependency. mesos uses the  `shaded-protobuf` one. Let me do some tests with different hadoop profiles first

Reply:
Thanks for looking into it!